app/components/thumbnail.jsx

```diff
@@ -47,9 +47,10 @@ export default class Thumbnail extends React.Component {
     };
 
     if (this.props.format === 'mp4') {
+      const width = this.props.width < 999 ? this.props.width : '300';
       return (
         <div>
-          <video width="300" controls onClick={this.playVideo}>
+          <video width={width} controls={this.props.controls} onClick={this.playVideo}>
```

Comment:
Maybe pass `this.props.height` to the video element as well as width, and set the same styles for max width and height as are set for images.